### DISCUSSION TITLE
Can't Run Tts Generation Webui

### DISCUSSION THREAD

**Discussion by SyamsQ** [rsxdalv/TTS-WebUI#425] [Q&A]
I can't run Tts Generation Webui. How can I fix it?

![image](https://github.com/user-attachments/assets/78d94730-eeee-4397-b445-fbd0ff4ae619)
![image](https://github.com/user-attachments/assets/8adf67a1-1cca-4f7a-a4e6-cbb76139ace0)

**Comment by rsxdalv** [comment]
Hi, thanks for the report. I will try to find the reason why it tries to install in the wrong directory. Although it could be because you are using the G drive, I believe it was working fine before on any drive.

**Comment by rsxdalv** [comment]
Ok, I see what could be the reason - you are running it as an administrator. Is there a reason for that? Did you try running it as a non-admin first? In windows when running something as an administrator, it often resets the folder to Windows-System32 which is NOT where you want to install stuff.

**Comment by ramesharma372-hub** [comment]
Before uninstalling everything, I'd try to identify **where the failure occurs**, because "can't run generation" can be caused by several different things in TTS-WebUI.

### 1. Check the console, not just the WebUI

One thing that confuses many new users is that the WebUI sometimes appears to do nothing while the actual error is printed only in the terminal/console window.

Common examples include:

* Missing model downloads
* Hugging Face download failures
* Extension installation problems
* CUDA/PyTorch errors
* Permission issues inside Docker

Several users have reported situations where the UI appeared idle while model downloads or errors were only visible in the terminal output.

---

### 2. Verify the required extension is installed

Many models in TTS-WebUI are implemented as extensions.

Symptoms:

* Generate button does nothing
* Model appears selectable
* No audio is produced

Make sure the extension for that specific model is installed and loaded. Users have reported confusion because the base UI can show models that require additional components before generation works.

---

### 3. If using Docker, check persistence

A frequent Docker issue is:

* Model downloads disappear after restarting containers
* Installed packages vanish
* Generation works once and then stops

Verify that your cache and Hugging Face directories are mapped to persistent volumes. Multiple users reported that downloaded assets were lost between container restarts when caches were not persisted.

Example environment variables commonly used:

```yaml
environment:
  - HF_HOME=/app/cache
  - HUGGINGFACE_HUB_CACHE=/app/cache/hub
  - XDG_CACHE_HOME=/app/cache
```

---

### 4. Check Hugging Face downloads

If generation never starts:

* Open the Hugging Face cache location
* Confirm model files actually finished downloading
* Look for interrupted downloads

Some users encountered repeated download failures that prevented models from ever becoming usable.

---

### 5. Verify GPU detection

Check:

* CUDA version
* PyTorch CUDA support
* VRAM availability

Inside TTS-WebUI, verify the GPU information panel shows your GPU correctly.

If VRAM is insufficient, some models may fail silently or fall back to CPU, resulting in extremely slow generation.

---

### 6. Share these details

To help diagnose the problem, please post:

* Operating system (Windows/Linux/macOS)
* Installation type (One-click, Manual, Docker)
* GPU model
* TTS-WebUI version
* Model you're trying to run
* Full console error log
* Whether the model files finished downloading

Without the console log, people can only guess.

### My recommendation

If you're a first-time user, start with a lightweight model such as Kokoro and verify that audio generation works end-to-end before trying larger models like DIA, AceStep, Fish Speech, or VibeVoice. Once a simple model works, it's much easier to isolate whether the issue is model-specific or a general installation problem.
